Library/Formula/slimerjs.rb

```diff
@@ -1,3 +1,23 @@
+class FirefoxDependency < Requirement
+  fatal true
+  cask "Firefox"
+  download "https://www.mozilla.org/en-US/firefox/new"
+
+  satisfy { firefoxappdira? || firefoxappdirb? }
```

Comment:
Or make a single, static method called `firefox_app` which returns the location if it exists and `nil` if it does not. This should mean you can use it in `satisfy` and `caveats` below.